test/src/kbasesearchengine/test/events/handler/WorkspaceEventHandlerTest.java

```diff
@@ -1472,4 +1473,123 @@ public void newVersionEventWithOverwriteExistingData() throws Exception {
         Assert.assertEquals(expectedEvent, updatedEvent);
     }
 
+    /*
+    Test that a NEW_VERSION object event with the isPublic flag set to false will be updated
+    with the isPublic flag to true if the current state of the associated workspace
+    has the public permission field set to "r".
+    This triggers the isPublic logic in WorkspaceEventHandler.updateObjectEvent. 
+    */
+    @Test
+    public void newVersionPublicFlagUpdatedFromWorkspaceCase1() throws Exception {
+        final CloneableWorkspaceClient clonecli = mock(CloneableWorkspaceClient.class);
+        final WorkspaceClient cloned = mock(WorkspaceClient.class);
+        final WorkspaceClient wscli = mock(WorkspaceClient.class);
+        when(clonecli.getClientClone()).thenReturn(cloned);
+        when(clonecli.getClient()).thenReturn(wscli);
+
+        StatusEvent event = StatusEvent.getBuilder(
+                StorageObjectType.fromNullableVersion("WS", "foo", 1),
+                Instant.ofEpochMilli(20000),
+                StatusEventType.NEW_VERSION)
+                .withNullableOverwriteExistingData(true)  // overwrite existing data
+                .withNullableisPublic(Boolean.FALSE)
+                .withNullableAccessGroupID(3)
+                .withNullableObjectID("1")
+                .withNullableVersion(1)
+                .build();
+
+        List<Tuple11<Long, String, String, String,
+                Long, String, Long, String, String, Long, Map<String, String>>> objList = new ArrayList<Tuple11<Long, String, String, String, Long, String, Long, String, String, Long, Map<String,String>>>();
+        objList.add(objTuple(1L, null, "sometype", "date", 1L, "copier",
+                3L, "wsname", "checksum", 44, Collections.emptyMap()));
+
+        when(wscli.administer(any()))
+                // for deleted objects check
+                .thenReturn(new UObject(objList))
+                // for name check
+                .thenReturn(new UObject(new GetObjectInfo3Results().withInfos(objList)))
+                // for isPublic check
+                .thenReturn(new UObject(wsTuple(3L, "wsname", "username", "date", 7, "n", "r",
+                        "unlocked", Collections.emptyMap())));
+
+        final WorkspaceEventHandler weh = new WorkspaceEventHandler(clonecli);
+
+        StatusEvent updatedEvent = weh.updateObjectEvent(event);
+
+        StatusEvent expectedEvent = StatusEvent.getBuilder(
+                StorageObjectType.fromNullableVersion("WS", "foo", 1),
+                Instant.ofEpochMilli(20000),
+                StatusEventType.NEW_VERSION)
+                .withNullableOverwriteExistingData(true)  // overwrite existing data
+                .withNullableisPublic(Boolean.TRUE)
+                .withNullableAccessGroupID(3)
+                .withNullableObjectID("1")
+                .withNullableVersion(1)
+                .build();
+
+        Assert.assertNotSame("expected same event object after update", event, updatedEvent);
```

Comment:
comment should be "expected different event object after update"

Reply:
I've actually removed this assertion from the upcoming change to this file - it is rather pointless. It is actually copied from the test above. It seems to me too general an assertion to make - yes they are different but the assertion below it which indicates that the event should have been updated in a certain way is both more specific and also covers the case of it being different.